src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/PickaxeOfVeinMining.java

```diff
@@ -60,9 +61,15 @@ private void breakBlocks(Player p, List<Block> blocks, int fortune) {
         for (Block b : blocks) {
             if (SlimefunPlugin.getProtectionManager().hasPermission(p, b.getLocation(), ProtectableAction.BREAK_BLOCK)) {
                 b.getWorld().playEffect(b.getLocation(), Effect.STEP_SOUND, b.getType());
+                
+                if (getItem().containsEnchantment(Enchantment.SILK_TOUCH)) {
```

Comment:
getItem() will return the original SlimefunItemStack, not the currently used item!